source/common/logger/ur_logger.h

```diff
@@ -0,0 +1,104 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: MIT
+
+#include <cstdarg>
+#include <memory>
+
+#include "ur_sinks.h"
+
+namespace logger
+{
+
+enum class Level {
+    DEBUG,
+    INFO,
+    WARNING,
+    ERROR,
+    FATAL,
+    QUIET
+};
+
+
+class Logger
+{
+public:
+    Logger() {
+        this->sink = std::make_shared<sink::Sink>();
+    }
+
+    Logger(Level level) : level(level), sink() {
+        this->sink = std::make_shared<sink::Sink>();
+    }
+
+    ~Logger() = default;
+
+    void setLevel(Level level) {
+        this->level = level;
+    }
+
+    void logDebug(const char *format, ...) {
```

Comment:
hm, this function is essentially repeated n times for each of the enum parameters. You could create a single template function (or a normal function that accepts the enum parameter...) and use it for every variant.

Reply:
I wanted the usage of logging to be as simple as can be, thus preferring debug("...") over log(DEBUG, "..."). Shortened functions a bit, though.